app/src/main/java/com/example/myapplication/CredentialsManager.kt

```diff
@@ -0,0 +1,50 @@
+package com.example.myapplication
+
+class CredentialsManager {
+
+    private val emailPattern = ("[a-zA-Z0-9\\+\\%\\-\\+]{1,256}" +
+            "\\@" +
+            "[a-zA-Z0-9][0-zA-Z0-9\\-]{0,64}" +
+            "(" +
+            "\\." +
+            "[a-zA-Z0-9][0-zA-Z0-9\\-]{0,25}" +
+            ")+").toRegex()
+
+    fun isValidEmail(email: String): Boolean {
+        return email.matches(emailPattern)
+    }
+
+    fun isValidPassword(password: String): Boolean {
+        return password.length >= 8
+    }
+
+    fun validateCredentials(email: String, password: String, isCheckboxChecked: Boolean): Boolean {
```

Comment:
what is isCheckboxChecked?

Reply:
It checks whether the user ticked the checkbox before pressing the button " Next > "